### PR TITLE
Update Keras documentation

### PR DESCRIPTION
**Pull request by adriangb**
- Pass `loss` to the constructor as documented in https://www.adriangb.com/scikeras/stable/migration.html
- Update package versions
- Update syntax to use SciKeras compiled models

### CODE REVIEW

File: docs/source/keras.rst [hunk 1]
```diff
@@ -14,8 +14,8 @@ these packages need to be installed:
 
 .. code-block:: bash
 
-   $ pip install tensorflow>=2.3.0
-   $ pip install scikeras>=0.1.8
+   $ pip install tensorflow>=2.4.0
+   $ pip install scikeras>=0.3.2
```
Comment (stsievert) [inline docs/source/keras.rst:18]: Why is this required? Dask-ML will still work with the removed versions, right?

File: docs/source/keras.rst [hunk 2]
```diff
@@ -36,24 +36,18 @@ normal way to create a `Keras Sequential model`_
    from tensorflow.keras.layers import Dense
    from tensorflow.keras.models import Sequential
 
-   def build_model(lr=0.01, momentum=0.9):
```
Comment (stsievert) [inline docs/source/keras.rst:39]: :+1:

**Review by stsievert — COMMENTED** [review]
Looks like the tests are failing because TF/Keras isn't installed on this branch. Could you comment out this line to make sure tests pass?

https://github.com/dask/dask-ml/blob/0ea276da1d78db582f40e1c256dfca4f70e6cbc6/ci/posix.yaml#L53

The example on this doc page is tested in `tests/model_selection/test_keras.py` . Can you update that test too?

File: docs/source/keras.rst (hunk 1 above)
Comment (adriangb) [reply docs/source/keras.rst:18]: This should work generally, but some of the syntax in this tutorial may not. I think we should either update the versions, or remove them altogether (since not specifying a version usually gets you the latest version anyway).

Comment (stsievert) [reply docs/source/keras.rst:18]: Dask-ML will not work with SciKeras v0.1.7. I think that version didn't have serialization (?).

We should make a note about the versioning. "The example below uses X. The usage with lower versions may be different than this example."

Comment (stsievert) [reply docs/source/keras.rst:18]: Wasn't there also some issue about serialization of stateful optimizers like Adam?

**Comment by adriangb** [comment]
The tests in e73daa9 were failing within TF.
I cut out all of the dask stuff in 1591be8 and just did `model.fit(X, y).score(X, y)` and am getting no errors.
Any ideas why tests might be failing in e73daa9 @stsievert ([failure line](https://dev.azure.com/dask-dev/dask/_build/results?buildId=2111&view=logs&j=34469798-9dec-5482-c96c-ad1eda9ec58d&t=b5370614-32af-516a-8930-943e3caa82ec&l=941))?

File: docs/source/keras.rst (hunk 1 above)
Comment (stsievert) [reply docs/source/keras.rst:18]: That's a good reason to require SciKeras v0.3.0.
